### ISSUE TITLE
Support as Roslyn analyzer.

### ISSUE THREAD

**Issue by martasp** [jjrdk/ArchiMetrics#34]
Is this provide metrics as a warning? it could be written as a Roslyn analyzer.

**Comment by payelh** [comment]
Yes, you can add this to Roslyn Analyzer, and set the severity to DiagnosticSeverity.Error.  Then you will see those as a warning after compiling.

**Comment by jjrdk** [comment]
I guess it could be included in a Roslyn analyzer. I'd be happy to get a PR for this ;-)

Just be aware that there have been a lot of new language features, which have not been included into the analyses here.
